engine/runtime-integration-tests/src/test/java/org/enso/interpreter/test/scope/ModuleScopeTest.java

```diff
@@ -65,40 +47,162 @@ public void staticMethodIsInResolvedExports() throws IOException {
   }
 
   @Test
-  public void staticMethodIsRegisteredInModuleScope() throws IOException {
+  public void moduleMethodIsRegisteredInModuleScope() throws IOException {
+    var mainSrc =
+        Source.newBuilder(
+                LanguageInfo.ID, """
+        module_method _ = 42
+        """, "test.enso")
+            .build();
+    // ModuleScope is populated in IrToTruffle - at runtime. So we have to evaluate
+    // the main module before we inspect the ModuleScope.
+    var mainMod = ctxRule.eval(mainSrc);
+    var mainRuntimeMod = (Module) ctxRule.unwrapValue(mainMod);
+    var assocType = mainRuntimeMod.getScope().getAssociatedType();
+    assertThat(assocType, is(notNullValue()));
+    var moduleMethod = mainRuntimeMod.getScope().getMethodForType(assocType, "module_method");
+    assertThat(moduleMethod, is(notNullValue()));
```

Comment:
- I'd _assert function schema_ rather than just a simple `assertNotNull` check
- the schema assert would show us the `self` argument position
- that's an important information